lib/topology/platforms/shell.py

```diff
@@ -533,12 +555,28 @@ def connect(self, connection=None):
             raise AlreadyConnectedError(connection)
 
         # Create a child process
+        self._spawn_args.update(
```

Comment:
One reason was because https://github.com/HPENetworking/topology/pull/35/commits/94a84272a8a2064f0931bef3d8f5e0d7947cc3b1#diff-a537e2036a51aca1aa7487b0c3d8654eR562
was assigning to a variable the output of ``update`` which is ``None``.